Classes/YIPopupTextView.m

```diff
@@ -592,22 +590,21 @@ - (void)_changePopupViewFrameWithNotification:(NSNotification*)notification
                 break;
             case UIInterfaceOrientationLandscapeLeft:
                 // keyboard at portrait-right
-                popupViewHeight = keyboardRect.origin.x - bgOrigin.x - topMargin;
+                popupViewHeight = keyboardRect.origin.y - bgOrigin.y - topMargin;
                 break;
             case UIInterfaceOrientationLandscapeRight:
                 // keyboard at portrait-left
-                popupViewHeight = bgOrigin.x - keyboardRect.origin.x - keyboardRect.size.width - topMargin;
+                popupViewHeight = keyboardRect.origin.y -  bgOrigin.y - topMargin;
```

Comment:
Just an observation, bgOrigin always seem to be zero... But left it maintain prior logic.